shortfin/python/shortfin_apps/sd/components/builders.py

```diff
@@ -142,21 +154,38 @@ def needs_update(ctx):
     return False
 
 
-def needs_file(filename, ctx):
-    out_file = ctx.allocate_file(filename).get_fs_path()
+def needs_file(filename, ctx, namespace=FileNamespace.GEN):
+    out_file = ctx.allocate_file(filename, namespace=namespace).get_fs_path()
     if os.path.exists(out_file):
         needed = False
     else:
-        filekey = f"{ctx.path}/{filename}"
+        name_path = "bin" if namespace == FileNamespace.BIN else ""
+        if name_path:
+            filename = os.path.join(name_path, filename)
+        filekey = os.path.join(ctx.path, filename)
         ctx.executor.all[filekey] = None
         needed = True
     return needed
 
 
+def needs_compile(filename, target, ctx):
+    device = "amdgpu" if "gfx" in target else "llvmcpu"
+    vmfb_name = f"{filename}_{device}-{target}.vmfb"
+    namespace = FileNamespace.BIN
+    return needs_file(vmfb_name, ctx, namespace)
+
+
+def get_cached_vmfb(filename, target, ctx):
+    device = "amdgpu" if "gfx" in target else "llvmcpu"
+    vmfb_name = f"{filename}_{device}-{target}.vmfb"
+    namespace = FileNamespace.BIN
+    return ctx.file(vmfb_name)
+
+
 @entrypoint(description="Retreives a set of SDXL submodels.")
 def sdxl(
     model_json=cl_arg(
-        "model_json",
+        "model-json",
```

Comment:
As a followup, let's be consistent on underscore or dash flags throughout the sd directory.